next permutation in c++

```diff
@@ -0,0 +1,69 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+//to print a vector
+void print(vector<int> a,int n){ 
+	for(int i=0;i<n;i++)
+		cout<<a[i]<<" ";
+	cout<<endl;
+}
+
+void nextPermutation(vector<int> a,int n){
+	int k=-1,l,temp;
+	
+	//finding largest k s.t. a[k]<a[k+1]
+	for(int i=0;i<n-1;i++){
+		if(a[i]<a[i+1])
+			k=i;
+	}
+	
+	//if k not updated sort  and print
+	if(k==-1){
+		sort(a.begin(),a.end());
+		print(a,n);
+		return;
+	}
+
+	//find the largest l s.t. a[k]<a[l]
+	for(int i=k+1;i<n;i++){
+		if(a[i]>a[k])
+		l=i;
+	}
+
+	//swap a[k] and a[l]
+	temp=a[k];
+	a[k]=a[l];
+	a[l]=temp;
+	
+	//print upto a[k]
+	for(int i=0;i<=k;i++)
+		cout<<a[i]<<" ";
+	
+	//reverse printing for a[k+1] to a[n-1] 
+	for(int i=n-1;i>k;i--)
+		cout<<a[i]<<" ";
+	
+	cout<<endl;
+	return;
+}
+
+int main(){
+	int n,item;
+
+	cout<<"enter length of permutation\n";
+	scanf("%d",&n);
```

Comment:
Suggestion for a better command-line user interface: As `std::scanf()` returns the number of successfully scanned items it allows you to distinguish parsing errors/end-of-input from correct inputs.
So you can have a `while` loop and keep pushing to the vector until you get zero elements.

For more info on `std::scanf()` look here: https://en.cppreference.com/w/cpp/io/c/fscanf
That link also has a duggestion on how to deal with input locales. While not relevant to this example, consider what happens if you want to enter floating point numbers and are on, say, a German system where the decimal and the thousand separators are switched compared to US English.